task/06-conditions-n-loops-tasks.js

```diff
@@ -251,7 +276,33 @@ function reverseInteger(num) {
  *   4916123456789012 => false
  */
 function isCreditCardNumber(ccn) {
-    throw new Error('Not implemented');
+    let parity = (String(ccn).split('').length %2 == 0);
+    let arr = ('' + ccn).split('').map(function(digit)  {return + digit;});
+     
+    function changeItem(item) {
+        item = item *2;
+        if (item > 9) {
+            item = item -9;
+        }
+        return item;
+    }
+
+    let sum = arr.reduce(function(total, current, index) {
+        if ((index - 1 == 0) && parity) {
+            total = changeItem(total);
+        }
+
+        if ((index %2 == 0) && parity) {
+            current = changeItem(current);
+        }
+
+        if ((index %2 != 0) && !parity) {
```

Comment:
"operator !== is always preferable
sometimes ''!=' may cause an error
(check at console :
'0' != 0
and '0' !== 0)"